src/big/big_float.cr

```diff
@@ -17,6 +17,18 @@ struct BigFloat < Float
     LibGMP.mpf_init_set_str(out @mpf, str, 10)
   end
 
+  def initialize(num : BigInt)
```

Comment:
This pull request is much broader than simply changing the hashing calculation. But these updates are really important.